Sources_v3/Workers/EventSender_Tests.swift

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChatClient
+import XCTest
+
+class EventSender_Tests: StressTestCase {
+    typealias ExtraData = DefaultDataTypes
+    
+    var webSocketClient: WebSocketClientMock!
+    var apiClient: APIClientMock!
+    var database: DatabaseContainerMock!
+    var time: VirtualTime!
+    var eventSender: EventSender<ExtraData>!
+    
+    override func setUp() {
+        super.setUp()
+        
+        webSocketClient = WebSocketClientMock()
+        apiClient = APIClientMock()
+        database = try! DatabaseContainerMock(kind: .inMemory)
+        
+        time = VirtualTime()
+        VirtualTimeTimer.time = time
+        
+        eventSender = EventSender(database: database, webSocketClient: webSocketClient, apiClient: apiClient)
+        eventSender.timer = VirtualTimeTimer.self
+    }
+    
+    override func tearDown() {
+        apiClient.cleanUp()
+        super.tearDown()
+    }
+    
+    func test_keystroke_startTypingAndStopAfterTimeout() throws {
+        // Send keystroke.
+        let cid = ChannelId.unique
+        eventSender.keystroke(in: cid)
+        
+        var referenceEndpoint: Endpoint<EventPayload<ExtraData>> = .event(cid: cid, eventType: .userStartTyping)
+        XCTAssertEqual(apiClient.request_endpoint, AnyEndpoint(referenceEndpoint))
+        apiClient.request_endpoint = nil
+        
+        // Wait timeout.
+        time.run(numberOfSeconds: .startTypingEventTimeout)
+        
+        // Check stop typing did send.
+        referenceEndpoint = .event(cid: cid, eventType: .userStopTyping)
+        XCTAssertEqual(apiClient.request_endpoint, AnyEndpoint(referenceEndpoint))
+    }
```

Comment:
I think it's better to use `apiClient.request_allRecordedCalls` for all these tests instead of resetting the `apiClient.request_endpoint` value.